keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/CombinedRepository.kt

```diff
@@ -81,26 +81,31 @@ class CombinedRepository(
 
   @Transactional(propagation = REQUIRED)
   override fun upsertDeliveryConfig(deliveryConfig: DeliveryConfig): DeliveryConfig {
-    val old = try {
+    val configWithSameName = try {
       getDeliveryConfig(deliveryConfig.name)
     } catch (e: NoSuchDeliveryConfigException) {
       null
     }
 
-    val existingConfig = try {
+    if (configWithSameName != null && configWithSameName.application != deliveryConfig.application) {
+      // we don't allow storing 2 configs with the same name, for different applications
+      throw DuplicateDeliveryConfigsException(configWithSameName.application)
```

Comment:
I think a better name for this exception might be `ConflictingDeliveryConfigsException`.